scala-package/core/src/main/scala/org/apache/mxnet/NDArray.scala

```diff
@@ -509,6 +510,63 @@ object NDArray extends NDArrayBase {
     array(sourceArr, shape, null)
   }
 
+  /**
+    * Create a new NDArray based on the structure of source Array
+    * @param sourceArr Array[Array...Array[MX_PRIMITIVE_TYPE]...]
+    * @param ctx context like to pass in
+    * @return an NDArray with the same shape of the input
+    */
+  def toNDArray(sourceArr: Array[_], ctx : Context = null) : NDArray = {
+    val shape = ArrayBuffer[Int]()
+    shapeGetter(sourceArr, shape, 0)
+    val container = new Array[Any](shape.product)
+    arrayCombiner(sourceArr, container, 0, container.length - 1)
+    val finalArr = container(0) match {
+      case f: Float => array(container.map(_.asInstanceOf[Float]), Shape(shape), ctx)
+      case d: Double => array(container.map(_.asInstanceOf[Double]), Shape(shape), ctx)
+      case _ => throw new IllegalArgumentException(s"Unsupported type ${container(0).getClass}")
```

Comment:
Should we add to the method docs as well that it will throw in an IllegalArgumentException ?

Reply:
Should the exception message indicate that the supported types are Float/Double ?